### PR TITLE
LIG Uplink ICM types & enclosure index support

### PR DESCRIPTION
**Pull request by vranystepan**
### Description
Added support for uplink ports with specified `:type` and `:enclosure_index` as mentioned in #225 

### Issues Resolved
- #225 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [x] Functional tests with the real Synergy environment
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by vranystepan** [comment]
Functional test with the [example](https://github.com/vranystepan/oneview-chef/blob/lig_uplink_icm_types/examples/logical_interconnect_group_synergy.rb) works. 

**Result:** 

result

File: examples/logical_interconnect_group_synergy.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+# Logical Interconnect Group 1
+oneview_logical_interconnect_group 'LogicalInterconnectGroup1' do
+  client my_client
+  data({
+     redundancyType: 'HighlyAvailable',
+     interconnectBaySet: 3,
+     enclosureIndexes: [1, 2],
+     enclosureType: 'SY12000'
+  })
```
Comment (tmiotto) [inline examples/logical_interconnect_group_synergy.rb]: I guess these `{ }` are redundant.

When you pass `data( ... )` like that, without a space between the tag and parentheses, it automatically transforms the content into a hash.

**Comment by tmiotto** [comment]
@vranystepan  is this PR ready to be reviewed?

**Comment by vranystepan** [comment]
@tmiotto yes

File: libraries/resource_providers/api200/logical_interconnect_group_provider.rb [hunk 2]
```diff
@@ -37,7 +37,11 @@ def load_uplink_sets
                   end
             up.add_network(net)
           end
-          parsed_uplink_info[:connections].each { |link| up.add_uplink(link[:bay], link[:port]) }
+          parsed_uplink_info[:connections].each do |link|
+            enclosure_index = link[:enclosure_index] ? link[:enclosure_index] : 1
```
Comment (tmiotto) [inline libraries/resource_providers/api200/logical_interconnect_group_provider.rb]: You can simply use:
```ruby
 enclosure_index = link[:enclosure_index] || 1
```

File: libraries/resource_providers/api200/logical_interconnect_group_provider.rb [hunk 3]
```diff
@@ -37,7 +37,11 @@ def load_uplink_sets
...
+            type = link[:type] ? link[:type] : nil
```
Comment (tmiotto) [inline libraries/resource_providers/api200/logical_interconnect_group_provider.rb]: You can simply use:
```ruby
type = link[:type]
```

File: spec/fixtures/cookbooks/oneview_test_api300_synergy/recipes/logical_interconnect_group_load_resource_with_synergy_properties.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#
+# Cookbook Name:: oneview_test
+# Recipe:: logical_interconnect_group_load_resource_with_properties
```
Comment (tmiotto) [inline spec/fixtures/cookbooks/oneview_test_api300_synergy/recipes/logical_interconnect_group_load_resource_with_synergy_properties.rb]: Update the Cookbook name and the Recipe name

File: spec/unit/resources/logical_interconnect_group/load_resource_with_synergy_properties_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
+require_relative './../../../spec_helper'
+
+describe 'oneview_test_api300_synergy::logical_interconnect_group_load_resource_with_synergy_properties' do
```
Comment (tmiotto) [inline spec/unit/resources/logical_interconnect_group/load_resource_with_synergy_properties_spec.rb]: Please move this entire describe to the file `load_resource_with_properties_spec.rb`, and delete this file thereafter.

**Review by tmiotto — CHANGES_REQUESTED** [review]
Great job @vranystepan ! ✨ 

There are just few things to be changed, and also, please rebase your code and re-run the `rake test` again, some gems were updated and this code may be affected by these changes.

**Comment by vranystepan** [comment]
@tmiotto I've made these changes .... anyway (after the rebase) it seems to be a bit messy (51 changed files?? OMG what I've done :grin: ). Is it normal or should I start over?

**Comment by vranystepan** [comment]
PS I'm willing to create a new fork and new Pull Request ... that's not so big deal :)

**Comment by fgbulsoni** [comment]
@vranystepan with 63 commits and 51 files changed it does look a bit messy 😨 
Something likely went wrong during the rebase, otherwise just your commits should be listed.
If possible, my suggestion would be:
1. Create a new local branch based on the updated Master
2. Add your changes to it
3. Force push to this branch. 
No need to close/reopen the PR in that case, and it should be clean then.

**Comment by vranystepan** [comment]
@fgbulsoni Sounds great! Will do. Thank you very much for your advice :)

**Comment by vranystepan** [comment]
@tmiotto Ufff :sweat: .... it should be OK now. Could you review please?

**Review by tmiotto — APPROVED** [review]

**Comment by tmiotto** [comment]
@vranystepan  It's fine now! 😄 

Just resolve the conflicts in the CHANGELOG and :shipit:

**Pull request merged**